src/java/main/org/apache/zookeeper/server/PurgeTxnLog.java

```diff
@@ -133,11 +133,17 @@ public boolean accept(File f){
             }
         }
         // add all non-excluded log files
-        List<File> files = new ArrayList<File>(Arrays.asList(txnLog
-                .getDataDir().listFiles(new MyFileFilter(PREFIX_LOG))));
+        List<File> files = new ArrayList<File>();
+        File[] fileArray;
+        if ((fileArray = txnLog.getDataDir().listFiles(new MyFileFilter(PREFIX_LOG))) != null) {
```

Comment:
Can we keep the var assignment `File[] fileArray` outside along with the object reference instead of clubbing with if check.

 File[] fileArray = txnLog.getDataDir().listFiles(new MyFileFilter(PREFIX_LOG);